### PR TITLE
Use Ruby 2.1 native exception cause handling

### PR DESCRIPTION
**Pull request by dazza-codes**
This is a `git cherry-pick` from the `develop` branch commit in 87076e2d325928612dd851cb00c6085ed1aa06fc, which applied without changes. This could help to resolve conflicts between master..develop for #34.

### CODE REVIEW

File: lib/sdr/sdr_robot.rb [hunk 1]
```diff
@@ -26,21 +25,35 @@ def initialize(workflow_name, step_name, opts = {})
...
-          LyberCore::Log.error druid + " - " + e.message + "\n" + e.backtrace.join("\n")
-          update_workflow_error_status 'sdr', druid, self.class.workflow_name, self.class.step_name, e.message
-          return 'error'
+        elapsed = Benchmark.realtime do
+          perform druid # implemented in the robot subclass
+        end
+        update_workflow_status 'sdr', druid, self.class.workflow_name, self.class.step_name, 'completed', elapsed
+        LyberCore::Log.info "Completed #{druid} in #{elapsed} seconds"
+        return 'completed'
+      rescue Exception => e
```
Comment (jcoyne) [inline lib/sdr/sdr_robot.rb]: Don't get stabbed: http://www.zenspider.com/ruby/quickref.html#general-tips

How about `rescue StandardError` instead?

File: lib/sdr_ingest/transfer_object.rb [hunk 2]
```diff
@@ -109,7 +109,7 @@ def cleanup_deposit_files(druid, bag_pathname)
             GC.start
```
Comment (jcoyne) [inline lib/sdr_ingest/transfer_object.rb]: 🤦‍♂️

**Review by jcoyne — DISMISSED** [review]

File: lib/sdr/sdr_robot.rb (hunk 1 above)
Comment (darrenleeweber) [inline lib/sdr/sdr_robot.rb]: Replaced all.

File: lib/sdr_ingest/transfer_object.rb (hunk 2 above)
Comment (darrenleeweber) [inline lib/sdr_ingest/transfer_object.rb]: Removed all.

**Review by jcoyne — APPROVED** [review]

**Comment by jcoyne** [comment]
I don't want to hold this up further, but there's a great gem called `retriable` that's perfect for the retry something n times use case.

**Pull request merged**